### PR TITLE
Fix a few issues catched by valgrind 

### PR DESCRIPTION
**Pull request by xhaihao**

### CODE REVIEW

File: _studio/mfx_lib/scheduler/src/mfx_scheduler_core_ischeduler.cpp [hunk 1]
```diff
@@ -623,6 +623,7 @@ mfxStatus mfxSchedulerCore::AddTask(const MFX_TASK &task, mfxSyncPoint *pSyncPoi
         m_pFreeTasks->param.sourceInfo.pFileName = pFileName;
         m_pFreeTasks->param.sourceInfo.lineNumber = lineNumber;
         // set the sync point for the task
+        handle.handle = 0;
```
[the remark this answers is not in the document]
Comment (xhaihao) [reply _studio/mfx_lib/scheduler/src/mfx_scheduler_core_ischeduler.cpp]: Updated, thanks.

File: _studio/mfx_lib/shared/src/mfx_h264_encode_vaapi.cpp [hunk 2]
```diff
@@ -2819,7 +2819,11 @@ mfxStatus VAAPIEncoder::Execute(
  /*
  *   RollingIntraRefresh
  */
-    if (memcmp(&task.m_IRState, &m_RIRState, sizeof(m_RIRState)))
+    if (task.m_IRState.refrType != m_RIRState.refrType ||
```
[the remark this answers is not in the document]
Comment (xhaihao) [reply _studio/mfx_lib/shared/src/mfx_h264_encode_vaapi.cpp]: Thanks for the comment, I introduces operator== for RIR state comparison in the new version

**Comment by dmitryermilov** [comment]
Thanks Haihao

**Comment by mgonchar** [comment]
@dvrogozh I'm ok with proposed solution. Maybe I'd rather define also ```operator !=```, but that is up to developer

**Pull request merged**